### PR TITLE
feature: create pipelines for docker image builds and releases

### PR DESCRIPTION
**Pull request by lmoesle**
**Issue**

#254 

**Description**

**Checklist**

* Code is easy to understand
* No obvious errors or bugs
* CI/CD Pipelines did run successfully
* Tests were implemented
* Documentation was created

### CODE REVIEW

File: .github/workflows/workflow-bump-version.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      - name: BUMP VERSION / Raise mvn version
+        if: inputs.new-maven-version != ''
+        run: |
+          ./mvnw versions:set --batch-mode -DprocessAllModules -DnewVersion=${{ inputs.new-maven-version }}-SNAPSHOT -DprocessAllModules
+          ./mvnw versions:commit -DprocessAllModules
+      - name: BUMP VERSION / Raise mvn version
+        if: inputs.new-maven-version == ''
+        run: |
+          ./mvnw build-helper:parse-version versions:set -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.nextMinorVersion}.0-SNAPSHOT -DprocessAllModules
+          ./mvnw versions:commit -DprocessAllModules
```
Comment (lmoesle) [inline .github/workflows/workflow-bump-version.yaml:46]: This overwrites the `${revision}` with the new version. How should we auto version bump maven after releases?

File: .github/workflows/workflow-build-and-release-services.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+          chmod +x ./mvnw
+          ./mvnw versions:set --batch-mode -DremoveSnapshot -DprocessAllModules
+      - name: RELEASE / Build and Release Maven
+        if: inputs.maven-release == true
+        uses: samuelmeuli/action-maven-publish@v1
+        with:
+          gpg_private_key: ${{ secrets.pgp_private_key }}
+          gpg_passphrase: ${{ secrets.pgp_private_key_secret }}
+          nexus_username: ${{ secrets.nexus_username }}
+          nexus_password: ${{ secrets.nexus_password }}
```
Comment (lmoesle) [inline .github/workflows/workflow-build-and-release-services.yml:69]: Should we use something else for maven releases?

Comment (dominikhorn93) [reply .github/workflows/workflow-build-and-release-services.yml:69]: i would use another one

File: .github/workflows/workflow-bump-version.yaml (hunk 1 above)
Comment (lmoesle) [reply .github/workflows/workflow-bump-version.yaml:46]: @DaAnda97 how do you usually do version bumps with the maven revision?

Comment (DaAnda97) [reply .github/workflows/workflow-bump-version.yaml:46]: the version usually is MAJOR.MINOR.PATCH
I used to to that manually yet

Comment (lmoesle) [reply .github/workflows/workflow-bump-version.yaml:46]: Should we stick to manually updating the version or should we use the auto version bump and overwrite all `${revision}`?

Comment (DaAnda97) [reply .github/workflows/workflow-bump-version.yaml:46]: I'd prefere an automatic increase

**Pull request merged**